src/Microsoft.TestPlatform.Common/DataCollection/Interfaces/IDataCollectionManager.cs

```diff
@@ -42,7 +42,7 @@ internal interface IDataCollectionManager : IDisposable
         /// <summary>
         /// Raises SessionStart event to all data collectors configured for run.
         /// </summary>
-        /// <returns>Are test case level events required.</returns>
+        /// <returns>bolean value specifying whether test case events are subscribed by datacollectors. Based on this execution process will decide whether to send TestCaseStart and TestCaseEnd events to dataCollector process.</returns>
```

Comment:
`Out of proc`
If no events are subscribed, then socket communication for communicating with Execution process is not established, dataCollectionPort value is returned as 0 to the vstest.console.exe process signifying that testcase level events are not required for this data collection session.

`In proc`
Don't think there would be a significant cost of sending the event for In Proc data collectors.